### ISSUE TITLE
terragrunt 0.67.2 claims terraform AWS S3 state bucket created with previous version not encrypted although SSE-KMS is enabled

### ISSUE THREAD

**Issue by mgerlach** [gruntwork-io/terragrunt#3384]
## Describe the bug

With terragrunt 0.67.2, running `terragrunt init` in a project with AWS S3 backend created by a previous terragrunt version with encryption enbaled (AWS console shows SSE-KMS active with a KMS key ARN, created with tg 0.67.1), the following prompt appears

```
23:28:12.935 WARN   The remote state S3 bucket xxx needs to be updated:
23:28:12.935 WARN     - Bucket Server-Side Encryption
Remote state S3 bucket xxx is out of date. Would you like Terragrunt to update it? (y/n)
```

When I enter 'y', the following appears

```
23:28:20.777 WARN   Encryption is not enabled on the S3 remote state bucket xxx. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!
23:28:20.777 INFO   Enabling Server-Side Encryption for the remote state AWS S3 bucket xxx.
23:28:21.328 INFO   Successfully enabled Server-Side Encryption for the remote state AWS S3 bucket xxx.
```

When I check the config in the AWS console, nothing has changed.

While this could be tolerated for manual interactions, it breaks automated workflows. In our case, GitHub Actions would just stop with unknown exit code:
```
Remote state S3 bucket inventory-quality-gateway-terraform-***-eu-central-1 is out of date. Would you like Terragrunt to update it? (y/n) 21:17:15.507 ERROR  EOF
21:17:15.507 ERROR  Unable to determine underlying exit code, so Terragrunt will exit with error code 1
Error: Process completed with exit code 1.
```

## Steps To Reproduce

Create the following files in your working dir:

terragrunt.hcl:
```hcl
remote_state {
  backend  = "s3"
  generate = {
    path      = "backend.tf"
    if_exists = "overwrite_terragrunt"
  }
  config = {
    bucket              = "test-tg-${get_aws_account_id()}"
    key                 = "terraform.tfstate"
    region              = local.aws_region
    dynamodb_table      = "test-tg-${get_aws_account_id()}-lock"
    encrypt             = true
  }
}
```

main.tf:
```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.0.0"
    }
  }
}

provider "aws" {
  region = "eu-central-1"
}
```

Then
- Install terraform 1.9.5 and terragrunt 0.67.1
- In your working dir, run `terragrunt init --terragrunt-non-interactive`
- In your working dir, run `terraform apply`

Then upgrade to terragrunt 0.67.2, delete the ./.terraform directory (but keep the generated backend.tf file!) and run `terragrunt init` again in your working dir.

## Expected behavior

No prompt for updating the state bucket / switching on server side encryption should appear when SSE is already enabled.

I would not want to add the --terragrunt-non-interactive flag to the automated execution.

## Versions

- Terragrunt version: 0.67.2
- OpenTofu/Terraform version: terraform 1.9.5
- Environment details: MacOSX Sonoma 14.5, Ubuntu 22.04 (GitHub Actions runner 2.319.1)

**Comment by buronary** [comment]
Same issue here. 
Our remote S3 Bucket is encrypted with SSE-KMS with a custom key. 
When I enter 'y'. it disables the Bucket Key and changes to SSE-KMS S3 default key

After that the terragrunt plan works again and shows the changes to enable the Bucket key and and change to KMS custom key

**Comment by mgerlach** [comment]
It our case the bucket encryption has already been using the default S3 KMS key ('aws/s3') forever... so we got prompted and upon entering 'y', nothing changed.

**Comment by yhakbar** [comment]
Hey folks

It seems like this issue came about due to unexpected side effects of this change:
https://github.com/gruntwork-io/terragrunt/pull/3383

At the very least, it appears that this bug fix was an unanticipated breaking change, and should not have been released as a patch. It also seems like introducing a change like will require a lot more testing to do safely.

I'll revert the change and re-open the instigating issue. We can look to revisit the issue in the future, but will communicate clearly that it is a breaking change when we do.

**Comment by yhakbar** [comment]
Fix released in [v0.67.3](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.67.3)
